pkg/operator/ceph/cluster/utils.go

```diff
@@ -32,26 +32,61 @@ import (
 // Its content allows modifying Ceph configuration flags
 func populateConfigOverrideConfigMap(clusterdContext *clusterd.Context, namespace string, ownerInfo *k8sutil.OwnerInfo) error {
 	ctx := context.TODO()
-	placeholderConfig := map[string]string{
-		k8sutil.ConfigOverrideVal: "",
+	// Add the helm labels in case the user wants to install the cluster helm chart to start managing it
+	labels := map[string]string{
+		"app.kubernetes.io/managed-by": "Helm",
 	}
-
-	cm := &v1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      k8sutil.ConfigOverrideName,
-			Namespace: namespace,
-		},
-		Data: placeholderConfig,
+	// Add the helm annotations in case the user wants to install the cluster helm chart to start managing it
+	annotations := map[string]string{
+		"meta.helm.sh/release-name":      "rook-ceph-cluster",
+		"meta.helm.sh/release-namespace": namespace,
 	}
 
-	err := ownerInfo.SetControllerReference(cm)
+	existingCM, err := clusterdContext.Clientset.CoreV1().ConfigMaps(namespace).Get(ctx, k8sutil.ConfigOverrideName, metav1.GetOptions{})
 	if err != nil {
-		return errors.Wrapf(err, "failed to set owner reference to override configmap %q", cm.Name)
-	}
-	_, err = clusterdContext.Clientset.CoreV1().ConfigMaps(namespace).Create(ctx, cm, metav1.CreateOptions{})
-	if err != nil && !kerrors.IsAlreadyExists(err) {
-		return errors.Wrapf(err, "failed to create override configmap %s", namespace)
+		if !kerrors.IsNotFound(err) {
+			logger.Warningf("failed to get cm %q to check labels and annotations", k8sutil.ConfigOverrideName)
+			return nil
+		}
+
+		// Create the configmap since it doesn't exist yet
+		placeholderConfig := map[string]string{
+			k8sutil.ConfigOverrideVal: "",
+		}
+		cm := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        k8sutil.ConfigOverrideName,
+				Namespace:   namespace,
+				Labels:      labels,
+				Annotations: annotations,
+			},
+			Data: placeholderConfig,
+		}
+
+		err := ownerInfo.SetControllerReference(cm)
+		if err != nil {
+			return errors.Wrapf(err, "failed to set owner reference to override configmap %q", cm.Name)
+		}
+		_, err = clusterdContext.Clientset.CoreV1().ConfigMaps(namespace).Create(ctx, cm, metav1.CreateOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to create override configmap %s", namespace)
+		}
+		logger.Infof("created placeholder configmap for ceph overrides %q", cm.Name)
+		return nil
 	}
 
+	// Add recommended labels and annotations to the existing configmap if it doesn't have any yet
+	if len(existingCM.Labels) == 0 && len(existingCM.Annotations) == 0 {
```

Comment:
i can understand why `and operator` is used as there wont be nay cluster with this prop,
but for durability, we can use `|| or operator`  as if someone added some settings manually

Reply:
Let's see if the next update will answer your question.